scripts/conf/treode.init

```diff
@@ -0,0 +1,161 @@
+#!/bin/sh
+# Copyright 2014 Treode, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### BEGIN INIT INFO
+# Provides:          treode
+# Required-Start:    $network $local_fs
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Treode server
+# Description:       Treode server
+### END INIT INFO
+
+# Author: treode@treode.com
+
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+
+#JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:/bin/java::")
+JAVA_HOME=/usr/lib/jvm/default-java
+DESC=treode            
+NAME=treode            
+DAEMON="$JAVA_HOME/bin/java"
+TREODEKV_DB=/opt/treode/db/store.3kv
+DAEMON_ARGS="-jar /opt/treode/lib/finagle-server-0.2.0.jar serve -solo  -log.output=/opt/treode/db/treode.log $TREODEKV_DB"
+PIDFILE=/var/run/$NAME.pid
+SCRIPTNAME=/etc/init.d/$NAME
+
+
+# Exit if the package is not installed
+[ -x $DAEMON ] || exit 0
+
+
+# Read configuration variable file if it is present
+[ -r /etc/default/$NAME ] && . /etc/default/$NAME
+
+# Load the VERBOSE setting and other rcS variables
+. /lib/init/vars.sh
+
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
+. /lib/lsb/init-functions
+
+#
+# Function that starts the daemon/service
+#
+do_start()
+{
+# Create 3KV_DB if does not exist
+    if [ ! -f $TREODEKV_DB ]; then
+	TREODEKV_HOST=`hostname -i | sed -e 's/\./ /g'|xargs printf '%02X'`
+	echo "Initialize Treode KV Store for host $TREODEKV_HOST"
+	$DAEMON -jar /opt/treode/lib/finagle-server-0.2.0.jar init -host=0x$TREODEKV_HOST -cell=0x$TREODEKV_HOST $TREODEKV_DB
+    fi
+
+	# Return
+	#   0 if daemon has been started
+	#   1 if daemon was already running
+	#   2 if daemon could not be started
+	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON --test > /dev/null \
+		|| return 1
+	start-stop-daemon --start --pidfile $PIDFILE --make-pidfile \
+	    --exec $DAEMON \
+	    --background -- \
+	    $DAEMON_ARGS \
+		|| return 2
+
+#	    --startas /bin/sh -- -c "exec $DAEMON $DAEMON_ARGS > /opt/treode/logs/treode.log 2>&1" \
```

Comment:
Similarly to the commented JAVA_HOME, this isn't helpful. Let's remove it, or explain what it's for.